alpine/alpine-infra/src/main/java/alpine/persistence/AlpineQueryManager.java

```diff
@@ -586,6 +585,9 @@ public List<Team> getTeams() {
         final Query<Team> query = pm.newQuery(Team.class);
```

Comment:
The method will need to return `PaginatedResult`, otherwise the API cannot provide the `X-Total-Count` header. You can simply use this `execute` method to cater to all this:

https://github.com/DependencyTrack/hyades-apiserver/blob/ad45a0ef8e3c2041945fc8dd1e1a489a8cc0f4fb/alpine/alpine-infra/src/main/java/alpine/persistence/AbstractAlpineQueryManager.java#L148-L162

I also think the frontend has a search box in the teams view. In that case we need to make sure that `filter` is being used (which is populated from the request's `searchText` query parameter). Something like this:

https://github.com/DependencyTrack/hyades-apiserver/blob/ad45a0ef8e3c2041945fc8dd1e1a489a8cc0f4fb/apiserver/src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java#L312-L316

Reply:
Does it find teams on a page that is isn't the current one? Say you have 26 teams named `A` to `Z`, you sort them by name and set the page size to `10`. Does it yield any results when you search for `Z`?

Reply:
Yes, like in SS below, `Team 8` is listed on second page and when I filter by `8` on first page, it appears.

<img width="799" alt="Screenshot 2025-06-23 at 13 38 10" src="https://github.com/user-attachments/assets/4d71aa3b-0b69-4153-80bb-1cc229d8de76" />
<img width="799" alt="Screenshot 2025-06-23 at 13 38 21" src="https://github.com/user-attachments/assets/d9673768-ea89-42b3-b206-9ec8eae3d2b1" />
<img width="799" alt="Screenshot 2025-06-23 at 13 38 34" src="https://github.com/user-attachments/assets/c4999853-2c1b-4a51-b211-cfdeb130e0d1" />

Reply:
It seems it's been done dynamically below : 
```
public PaginatedResult execute(final Query<?> query, final Object... parameters) {
        final long count = getCount(query, parameters);
        decorate(query);
        return new PaginatedResult()
                .objects(executeAndCloseWithArray(query, parameters))
                .total(count);
    }
```